lib/src/runner/live_suite_controller.dart

```diff
@@ -106,8 +106,12 @@ class LiveSuiteController {
   /// is `true` (the default), the test is put into [passed] if it succeeds.
   /// Otherwise, it's removed from [liveTests] entirely.
   ///
+  /// Likewise, if [countFailure] is `true` (the default), the test is put into
+  /// [failed] if it fails. Otherwise, it's removed from [liveTests] entirely.
```

Comment:
Another way to handle this would be to treat all retries as part of the same test, and just have that test not complete until all retries are complete—you'd probably handle that in the Invoker rather than the Engine.

If we decide to go with the `countFailures` approach, I think that means a breaking change to the JSON reporter protocol—it does have a notion of "hidden" tests currently, but the documentation guarantees that "only successful tests will be hidden" which won't be true after this. That indicates to me that a single-test approach would be less disruptive.